extensions.html

```diff
@@ -649,6 +649,168 @@ <h2>
+        <thead>
+          <tr>
+            <th scope="col">Buttons</th>
+            <th scope="col">Mapping</th>
+            <th scope="col">Type</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>buttons[0]</td>
+            <td>diamond_down</td>
+            <td>Standard: Bottom button in right cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[1]</td>
+            <td>diamond_right</td>
+            <td>Standard: Right button in right cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[2]</td>
+            <td>diamond_left</td>
+            <td>Standard: Left button in right cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[3]</td>
+            <td>diamond_up</td>
+            <td>Standard: Top button in right cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[4]</td>
+            <td>shoulder_left</td>
+            <td>Standard: Top left front button</td>
+          </tr>
+          <tr>
+            <td>buttons[5]</td>
+            <td>shoulder_right</td>
+            <td>Standard: Top right front button</td>
+          </tr>
+          <tr>
+            <td>buttons[6]</td>
+            <td>trigger_left</td>
+            <td>Standard: Bottom left front button</td>
+          </tr>
+          <tr>
+            <td>buttons[7]</td>
+            <td>trigger_right</td>
+            <td>Standard: Bottom right front button</td>
+          </tr>
+          <tr>
+            <td>buttons[8]</td>
+            <td>select</td>
+            <td>Standard: Left button in center cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[9]</td>
+            <td>start</td>
+            <td>Standard: Right button in center cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[10]</td>
+            <td>thumbstick_left</td>
+            <td>Standard: Left stick pressed button</td>
+          </tr>
+          <tr>
+            <td>buttons[11]</td>
+            <td>thumbstick_right</td>
+            <td>Standard: Right stick pressed button</td>
+          </tr>
+          <tr>
+            <td>buttons[12]</td>
+            <td>dpad_up</td>
+            <td>Standard: Top button in left cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[13]</td>
+            <td>dpad_down</td>
+            <td>Standard: Bottom button in left cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[14]</td>
+            <td>dpad_left</td>
+            <td>Standard: Left button in left cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[15]</td>
+            <td>dpad_right</td>
+            <td>Standard: Right button in left cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[16]</td>
+            <td>system</td>
+            <td>Standard: Center button in center cluster</td>
+          </tr>
+          <tr>
+            <td>buttons[17]</td>
```

Comment:
This suggests the "trackpad" button can only appear at `buttons[17]`. My thinking is that we should not continue the current pattern of assigning meaning to button indices, otherwise the `buttons` array will be very sparse since most gamepads do not share the same extra buttons. Instead of using button indices to identify buttons, apps can check the new enum attribute.

For this non-normative section, I think it will be clearer if we provide a separate table for each gamepad instead of trying to combine them all into a single table.

Reply:
@nondebug  Thanks for the review. Will update the PR based on the feedback shortly.

Reply:
@nondebug I have updated the PR based on your feedback. I have added the enum for identifying the gamepad buttons. Please review this changes and share your opinions.